src/Microsoft.AspNet.Razor.Runtime/TagHelpers/ReadOnlyTagHelperAttributes.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    public class ReadOnlyTagHelperAttributes<TAttributeType> : IReadOnlyList<TAttributeType>
```

Comment:
Will nix it. Just to follow up on confusion:

You're absolutely right, I mislead you. Looked into it more as to why it caused failures on my end and here's a detailed sample of how it fails to infer types:

```C#
public object Foo(IEnumerable attributes)
{
    return null;
}

public TValue Foo<TValue>(IEnumerable<TValue> attributes)
{
    return default(TValue);
}

var output = new TagHelperOutput(string.Empty);

// Calls the object Foo(IEnumerable attributes) overload because IEnumerable<TValue> cannot be inferred.
// Deleting the object Foo(IEnumerable) overload results in compilation error.
var attribute = Foo(output.Attributes); 
```

Came across the issue when attempting to use `Assert.Single` with the `ILookup` bit included.